packages/aws-rfdk/lib/lambdas/nodejs/pad-efs-storage/test/handlers.test.ts

```diff
@@ -0,0 +1,337 @@
+      await fsp.writeFile(filename, 'Some data');
+    }
+
+    // WHEN
+    // Add 2 10MB files to the temp directory.
+    await growFilesystem(2, 10, tempDirectory);
+
+    // THEN
+    // Make sure that the files that we added started numbering at 8
+    const dirContents = (await fsp.readdir(tempDirectory)).sort();
+    expect(dirContents).toEqual(['00008', '00009', '4', '5', '6', '7']);
+  });
+
+  test('Delete from directory', async () => {
+    // GIVEN
+    for (var i=1; i<8; i++) {
+      const filename = join(tempDirectory, i.toString().padStart(3, '0'));
+      await fsp.writeFile(filename, 'Some data');
+    }
+    const preDirContents = (await fsp.readdir(tempDirectory)).sort();
+
+    // WHEN
+    // Remove two files from the filesystem
+    await shrinkFilesystem(2, tempDirectory);
+
+    // THEN
+    const dirContents = (await fsp.readdir(tempDirectory)).sort();
+    expect(preDirContents).toEqual(['001', '002', '003', '004', '005', '006', '007']);
+    expect(dirContents).toEqual(['001', '002', '003', '004', '005']);
+  });
+});
+
+describe('Testing getDiskUsage behavior', () => {
+  var tempDirectory: string;
+
+  beforeEach(async () => {
+    // Create a temp directory for putting files.
+    tempDirectory = await fsp.mkdtemp(join(tmpdir(), 'tmp.'));
+  });
+  afterEach(async () => {
+    await recursiveDeleteDirectory(tempDirectory);
+    tempDirectory = '';
+  });
+
+  test('Throws when no mountPoint', async () => {
+    await expect(() => getDiskUsage({
+    },
+    {
+      logGroupName: '',
+      logStreamName: '',
+    })).rejects.toThrow();
+  });
+
+  test('Throws mountPoint does not exist', async () => {
+    await expect(() => getDiskUsage({
+      mountPoint: join(tempDirectory, 'does_not_exist'),
+    },
+    {
+      logGroupName: '',
+      logStreamName: '',
+    })).rejects.toThrow();
+  });
+
+  test('Throws when mountPoint not a directory', async () => {
+    // WHEN
+    const filename = join(tempDirectory, '001');
+    await fsp.writeFile(filename, 'Some data');
+
+    // THEN
+    await expect(() => getDiskUsage({
+      mountPoint: filename,
+    },
+    {
+      logGroupName: '',
+      logStreamName: '',
+    })).rejects.toThrow();
+  });
+
+  test('Correctly calculates disk usage', async () => {
+    // GIVEN
+    setDefaultFilesize(10);
+    const execPromise = promisify(exec);
+    await execPromise(`/usr/bin/dd if=/dev/zero of=${join(tempDirectory, 'file1.tmp')} bs=10MB count=1`);
+    await fsp.mkdir(join(tempDirectory, 'subdir'));
+    await execPromise(`/usr/bin/dd if=/dev/zero of=${join(tempDirectory, 'subdir', 'file2.tmp')} bs=10MB count=1`);
+
+    // WHEN
+    const usage = await getDiskUsage({
+      mountPoint: tempDirectory,
+    },
+    {
+      logGroupName: '',
+      logStreamName: '',
+    });
+
+    // THEN
+    expect(usage).toBe(2);
+  });
+
+});
+
+describe('Testing padFilesystem macro behavior', () => {
+
+  var tempDirectory: string;
+
+  beforeEach(async () => {
```

Comment:
Looks like we're using these `beforeEach` and `afterEach` functions for all of these `describe` blocks. Should we move these functions outside of the `describe` blocks (and the `tempDirectory` variable) so they only need to be defined once? Could also wrap all the `describe` blocks in one parent `describe` and put these functions in that parent scope

Reply:
Sure, works for me